flight/PiOS/STM32/chibi_main.c

```diff
@@ -82,10 +84,12 @@ int main()
 	/* Brings up System using CMSIS functions, enables the LEDs. */
 	PIOS_SYS_Init();
 	
-	/* For Sparky2 we use an RTOS task to bring up the system so we can */
-	/* always rely on an RTOS primitive */	
+#if !defined(PIOS_INCLUDE_CHIBIOS)
 	initTaskHandle = PIOS_Thread_Create(initTask, "init", INIT_TASK_STACK, NULL, INIT_TASK_PRIORITY);
 	PIOS_Assert(initTaskHandle != NULL);
+#else
+	initTask(NULL);
+#endif
 
 	while(1)
 		PIOS_Thread_Sleep(PIOS_THREAD_TIMEOUT_MAX);
```

Comment:
Maybe we can do system mod stuff in this thread in future. It's already not a normal module.

Reply:
Yup we have our plans for that in #1879, so that stacked with this would double our savings.